api/src/main/scala/docker/effect/interop/CovariantKleisli.scala

```diff
@@ -0,0 +1,99 @@
+package docker.effect
+package interop
+
+import cats.{ ApplicativeError, Monad, MonadError }
+
+import scala.language.implicitConversions
+
+sealed abstract class CovariantKleisli[F[+_]: Monad, -R, +A](private[interop] val rio: (=>R) => F[A]) {
```

Comment:
Same question

Reply:
No reason. Was to stay more general, but I guess there's not much optimization happening with this. I will change.